language-design.md

```diff
@@ -74,7 +230,27 @@ implementation choices from users?
 
 **Response**
 
-
+The choice of implementation for a DSL strongly affects its users. If we choose
+to implement an internal DSL, then users can have an easier time learning the
+language, because users may already be familiar with the host language. Even if
+they are not, the host language will have documentation, so it will likely be
+easier to pick up the DSL than if the DSL was a totally new language. On the
+other hand, if we implement our DSL as external, there is the advantage that we
+can choose syntax that makes sense to experts in our chosen domain. If those
+experts are our users, that will help them.
+
+In some cases it is not possible to hide implementation choices from users. For
+example, users will be able to tell whether our language is internal or
+external. However, whenever we can, it would be advisable to hide as much as
+possible. For example, if our language has a `sort` keyword, we do not need to
+tell the user more than how fast it is guaranteed to run. This is a principle
+advocated in [Bloch, 2006], as applied to APIs. We believe the same applies to
+DSLs. The main motivation here is that if we ever need to change implementation
+details, then there won't be false information lying around that we would have
+to track down and update, which would lead to confusion. We suspect that it
+depends on the context, but there is definitely a balance to be struck between
+leaving yourself wiggle room, and not annoying the users with too many black
+boxes.
```

Comment:
language design seems to have a lot of "finding the middle ground"

makes it super hard to do